post/tax.go

```diff
@@ -324,16 +324,37 @@ func (td *TaxDecorator) handleTaxableTransaction(
 		return err
 	}
 
+	params, err := td.feemarketKeeper.GetParams(ctx)
+	if err != nil {
+		return err
+	}
+
+	nativeDenom := params.FeeDenom
+
 	// Get fee payer and check if fee grant exists
 	tax := rewards.Add(burn...)
 	feePayer, err := td.getFeePayer(ctx, feeTx, tax, tx.GetMsgs())
 	if err != nil {
 		return err
 	}
 
-	// Deduct tax from fee payer
-	if err := td.deductTaxFromFeePayer(ctx, feePayer, tax); err != nil {
-		return err
+	// Check if fees contain only the native denom, if it is IBC denom fee abs logic applies.
+	// The fees will be deducted from the fee payer and the tokens already sent to did module account.
+	onlyNativeDenom := true
+	for _, fee := range feeTx.GetFee() {
+		if fee.Denom != nativeDenom {
+			// If any other token besides the native denom is present, set the flag to false
+			onlyNativeDenom = false
+			break
+		}
+	}
```

Comment:
This block might be better abstracted away to either a utility function or a method to the struct implementation.

e.g. `isOnlyNativeDenom`

It'll most definitely be reusable and more readable as symbolically action-based.